From: Shengjing Zhu <zhsj@debian.org>
Date: Wed, 7 Oct 2020 01:01:05 +0800
Subject: ignore bind error in embed/serve_test.go

In autopkgtest-go, this test will be run when etcd is running.
etcd-server is installed, then the service is started by default.

Forwarded: not-needed
---
 server/embed/serve_test.go | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/server/embed/serve_test.go b/server/embed/serve_test.go
index 6a19911..7db0435 100644
--- a/server/embed/serve_test.go
+++ b/server/embed/serve_test.go
@@ -17,6 +17,7 @@ package embed
 import (
 	"fmt"
 	"io/ioutil"
+	"net"
 	"net/url"
 	"os"
 	"testing"
@@ -49,7 +50,11 @@ func TestStartEtcdWrongToken(t *testing.T) {
 	cfg.AuthToken = "wrong-token"
 
 	if _, err = StartEtcd(cfg); err != auth.ErrInvalidAuthOpts {
-		t.Fatalf("expected %v, got %v", auth.ErrInvalidAuthOpts, err)
+		if _, ok := err.(*net.OpError); ok {
+			t.Skipf("got %v", err)
+		} else {
+			t.Fatalf("expected %v, got %v", auth.ErrInvalidAuthOpts, err)
+		}
 	}
 }
 
